### PR TITLE
Added relation to usage-monitor

### PR DESCRIPTION
**Pull request by AviadPepperi**

### CODE REVIEW

File: server-side/relations.usage.service.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            Description: "Health Monitor relation to Usage Monitor",
+            AddonRelativeURL: "/api/usage_callback",
+        }
+    };
+
+    async getUsageData(client: Client, request: Request) {
+        const today = new Date();
+        let size = 0
+        
+        // Get 3 latest entries
```
Comment (nofarpepperi) [inline server-side/relations.usage.service.ts:37]: get latest entry

File: server-side/relations.usage.service.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        const monthlyAddonUsageResponse = await this.papiClient.addons.data.uuid(this.clientData.addonUUID).table('DailyAddonUsage').iter({ order_by: "CreationDateTime desc", page_size: 1 }).toArray();
+
+        // Search for data from today
+        let todaysResult = monthlyAddonUsageResponse.find((dailyUsage) => {
```
Comment (nofarpepperi) [inline server-side/relations.usage.service.ts:41]: you are asking for page size = 1, you dont need to use find

File: server-side/relations.usage.service.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            const timeString = dailyUsage.CreationDateTime!
+            const time = new Date(timeString)
+            return time.toDateString() == today.toDateString()
+        })
+        
+        // If data not found take it from AWS
```
Comment (nofarpepperi) [inline server-side/relations.usage.service.ts:47]: add to comment // if there is no data of addon's usage in last day take it from cloud watch

File: server-side/relations.usage.service.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        if (todaysResult === undefined || (todaysResult as any).AddonsUsage === undefined) {
+            const monitorSettingsService = new MonitorSettingsService(client);
+            const distributor = await monitorSettingsService.papiClient.get('/distributor');
+            todaysResult = { AddonsUsage: await getCloudWatchLogs(monitorSettingsService, today.getTime(), distributor) }
```
Comment (nofarpepperi) [inline server-side/relations.usage.service.ts:51]: getCloudWatchLogs(service, now: number, distributor) is using only the distributor uuid, you have it on the token, there is no need to call to /distributor
also there is no need to send now parameter to getCloudWatchLogs, the function declare that "// Looking at data from one calendar day of yesterday", you can just use new Date().getTime() inside the function

File: server-side/relations.usage.service.ts (hunk 1 above)
Comment (AviadPepperi) [reply server-side/relations.usage.service.ts]: Gettings latest entry

File: server-side/relations.usage.service.ts (hunk 2 above)
Comment (AviadPepperi) [reply server-side/relations.usage.service.ts]: Changed find to if

File: server-side/relations.usage.service.ts (hunk 3 above)
Comment (AviadPepperi) [reply server-side/relations.usage.service.ts]: added

File: server-side/relations.usage.service.ts (hunk 4 above)
Comment (AviadPepperi) [reply server-side/relations.usage.service.ts]: Changed functions signatures

**Review by nofarpepperi — APPROVED** [review]

**Pull request merged**